core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -0,0 +1,73 @@
+module Spree
+  module UserAddressBook
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :user_addresses, foreign_key: "user_id", class_name: "Spree::UserAddress"
+
+      has_many :addresses, through: :user_addresses
+
+      has_one :default_user_address, -> { where default: true}, foreign_key: "user_id", class_name: 'Spree::UserAddress'
+      has_one :default_address, through: :default_user_address, source: :address
+
+      # bill_address is only minimally used now, but we can't get rid of it without a major version release
```

Comment:
I think having the billing address associated to a credit card makes some sense, however I can also think of some challenges in implementing that. It also has the drawback of having to pick a billing address for each credit card. Looking at the data from one existing store the majority of credit cards don't have an address_id on them (not sure when that association is created) so back-filling that can only be done if the card was used on an order. There may be some UI changes that stores will need to make to allow for selecting the billing address when adding a new credit card to a user. These are just some of the things that come to mind.

I don't have strong opinions about either option, but I think now is a good time to decide on how we want this to work in the long term and think about what will be involved in migrating from the current approach. Alternatively if decide we don't want to move the billing address to the credit card, I think there should be a way to specify the default billing address on a UserAddress.